genienlp/tasks/generic_dataset.py

```diff
@@ -1929,3 +1929,48 @@ def return_splits(cls, name, path='.data', train='train', validation='dev', test
         return Split(train=all_train_data, eval=all_validation_data, test=all_test_data), Split(
             train=train_path, eval=validation_path, test=test_path
         )
+
+
+class OODDataset(CQA):
+    name = 'ood'
+    is_sequence_classification = True
+
+    def __init__(self, path, lower=False, cached_path=None, skip_cache=False, **kwargs):
+        examples = []
+        question = 'Is this sentence in-domain or out-domain?'
+
+        cache_name = os.path.join(cached_path, os.path.basename(path))
+        if os.path.exists(cache_name) and not skip_cache:
+            logger.info(f'Loading cached data from {cache_name}')
+            examples = torch.load(cache_name)
+        else:
+            for fname in glob.iglob(os.path.join(path, '*.tsv')):
```

Comment:
What is the on-disk format now? Is it different directories for train/eval/test? Do we call it "dev" or "eval"?